src/tests/keywords/sha2

```diff
@@ -9,46 +9,63 @@ update {
        request:Tmp-Octets-0 := 0x000504030201
 }
 
+update {
+	request:Tmp-Octets-1 := "%{sha256:This is a string\n}"
+	request:Tmp-Octets-2 := "%{sha256:%{&Tmp-String-0}}"
+	request:Tmp-Octets-3 := "%{sha256:%{&request:Tmp-String-0}}"
+	request:Tmp-Octets-4 := "%{sha256:%{&request:Tmp-Octets-0}}"
+	request:Tmp-Octets-5 := "%{sha256:%{&request:Tmp-Octets-9}}"
+}
+
 #
 #  Put "This is a string" into a file and call "sha256sum" on it.
 #  You should get this string.
 #
-if ("%{sha256:This is a string\n}" != 'b3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124') {
+if (&request:Tmp-Octets-1 != 0xb3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124) {
 	test_fail
 }
 
-if ("%{sha256:&Tmp-String-0}" != 'b3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124') {
+if (&request:Tmp-Octets-2 != 0xb3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124) {
 	test_fail
 }
 
-if ("%{sha256:&request:Tmp-String-0}" != 'b3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124') {
+if (&request:Tmp-Octets-3 != 0xb3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124) {
 	test_fail
 }
 
-if ("%{sha256:%{Tmp-String-0}}" != 'b3716a1ab53042bb392034f29071e13b0c38aa19b4edd75d9a76022f91189124') {
+#
+#  SHA256 should also be able to cope with references to octet attributes
+#
+if (&request:Tmp-Octets-4 != 0xf307e202b881fded70e58017aa0c4d7b29c76ab25d02bf078301a5f6635187eb) {
 	test_fail
 }
 
 #
-#  SHA256 should also be able to cope with references to octet attributes
+# SHA256 of empty string
 #
-if ("%{sha256:&request:Tmp-Octets-0}" != 'f307e202b881fded70e58017aa0c4d7b29c76ab25d02bf078301a5f6635187eb') {
+if (&request:Tmp-Octets-5 != 0xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855) {
 	test_fail
 }
 
 #
 #  SHA512 and SHA256 share common code paths, so the tests don't need to be
 #  as exhaustive.
 #
-if ("%{sha512:This is a string\n}" != '56b57df5cce42d4e35c644649798ea23ec16f4f4626e78faf4d2d8f430ea349bcc28cd5532457c82f0aa66bf68988346039fe75b900a92ff94fd53993d45990f') {
+update {
+	request:Tmp-Octets-1 := "%{sha512:This is a string\n}"
+	request:Tmp-Octets-2 := "%{sha512:%{&Tmp-String-0}}"
+	request:Tmp-Octets-3 := "%{sha512:%{&request:Tmp-Octets-0}}"
```

Comment:
Mmm, those don't look right, shouldn't have attribute references with expansions like that.  I guess they probably still work because the tmpl code ignores the '&' but it should be removed.